### PR TITLE
Add new `Style/EmptyBlockParameter` cop

### PR DESCRIPTION
**Pull request by pocke**
This cop checks delimiters of empty block parameters.

```ruby
# bad
a do ||
end

# good
a do
end
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/empty_block_parameter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  it 'registers an offense for an empty block parameter with {} style' do
+    expect_offense(<<-RUBY.strip_indent)
+      a { || do_something }
+          ^^ Omit delimiters for the empty block parameters.
+    RUBY
+  end
+
+  it 'registers an offense for an empty block parameter with a lambda' do
+    expect_offense(<<-RUBY.strip_indent)
+      -> () { do_something }
```
Comment (Drenmi) [inline spec/rubocop/cop/style/empty_block_parameter_spec.rb]: I would much rather have this as a separate cop.

File: lib/rubocop/cop/style/empty_block_parameter.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      class EmptyBlockParameter < Cop
+        MSG = 'Omit delimiters for the empty block parameters.'.freeze
+
+        def_node_matcher :empty_arguments?, <<-PATTERN
+          (block _ $(args) _)
+        PATTERN
+
+        def on_block(node)
+          empty_arguments?(node) do |args|
+            return unless args.loc.expression
```
Comment (Drenmi) [inline lib/rubocop/cop/style/empty_block_parameter.rb]: Can we put this as a meaningfully named method in `ArgsNode`? 🙂

File: lib/rubocop/cop/style/empty_block_parameter.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for delimiters that are pipes or braces for empty
+      # block parameters. Delimiters for empty block parameters do not cause
+      # syntax errors, but it looks strange.
```
Comment (Drenmi) [inline lib/rubocop/cop/style/empty_block_parameter.rb]: "It looks strange" is a subjective valuation, I guess. Perhaps "are redundant"? 🙂

**Comment by Drenmi** [comment]
Nice! Didn't even know you could do this without syntax error. 😅

File: spec/rubocop/cop/style/empty_block_parameter_spec.rb (hunk 1 above)
Comment (pocke) [reply spec/rubocop/cop/style/empty_block_parameter_spec.rb]: 👍 I'll separate the cop into `Style/EmptyBlockParameter` and `Style/EmptyLambdaParameter`.

File: lib/rubocop/cop/style/empty_block_parameter.rb (hunk 2 above)
Comment (pocke) [reply lib/rubocop/cop/style/empty_block_parameter.rb]: Probably. I'll try it.

**Comment by bbatsov** [comment]
👍

**Pull request merged**